### ISSUE TITLE
Semantics excludeSemantics breaks aria-label on flutter web <input> for TextFormField

### ISSUE THREAD

**Issue by itsnits** [flutter/flutter#172206]
### Steps to reproduce

1. Create a TextFormField wrapped with a Semantics widget.
2. Add a label and set excludeSemantics: true.
3. Run on Flutter Web using flutter build web.
4. Try to inspect the generated <input> element.
5. Observe that aria-label, it will come as 'Login_email' which is provided in Semantics Label
6. Now set excludeSemantics: false and run the Flutter web.
7. Try to inspect the generated <input> element
8. Observe the aria-label, it will come as 'Email', which is the hint text provided in the TextFormField. 'Login_email' label is not coming in this case

Tested on Chrome stable version 138.0.7204.101

### Expected results

- The <input> should have aria-label="Login_email" so it can be used in web automation tools like Selenium/Cypress.

- Excluding the child semantics should still allow mapping the label (i.e 'Login_email' in my case) to the native input.

### Actual results

- Input cannot be typed into programmatically (via Playwright/Selenium/Cypress) in flutter web when excludeSemantics is set to true.

- If excludeSemantics: false, aria-label is coming as 'Email' (which is Hint text in InputDecoration of TextFormField, not the one which I am providing in labels of Semantics widget ) and input is interactive in flutter web. We can't use Hint text as semantics label as we are using language support and Hint Text can be changed based on selected language.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  runApp(const MyApp());
  Future.delayed(const Duration(seconds: 20), () {
    SemanticsBinding.instance.ensureSemantics();
  });
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Semantics Bug Demo',
      home: Scaffold(
        appBar: AppBar(title: const Text('Semantics Bug')),
        body: const Center(
          child: Padding(
            padding: EdgeInsets.all(16.0),
            child: LoginForm(),
          ),
        ),
      ),
    );
  }
}

class LoginForm extends StatefulWidget {
  const LoginForm({super.key});

  @override
  State<LoginForm> createState() => _LoginFormState();
}

class _LoginFormState extends State<LoginForm> {
  final _formKey = GlobalKey<FormState>();
  final TextEditingController _emailController = TextEditingController();
  final FocusNode _emailFocusNode = FocusNode();

  @override
  Widget build(BuildContext context) {
    return Form(
      key: _formKey,
      child: Semantics(
        label: 'Login_email',
        container: true,
        excludeSemantics: true,
        explicitChildNodes: true,
        textField: true,
        enabled: true,
        child: TextFormField(
          controller: _emailController,
          focusNode: _emailFocusNode,
          decoration: const InputDecoration(
            hintText: 'Email',
          ),
          validator: (value) {
            if (_emailFocusNode.hasFocus) return null;
            if (value == null || value.isEmpty) return 'Email is required';
            return null;
          },
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/9f8cde42-92ab-48c0-9540-340110ffcc48" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/3e69c290-e950-4b14-ada2-56434712cfe5" />

</details>

### Logs

<details open><summary>Logs</summary>

```console
Launching lib\main.dart on Chrome in debug mode...
This app is linked to the debug service: ws://127.0.0.1:63316/aE8KBBVWFXo=/ws
Debug service listening on ws://127.0.0.1:63316/aE8KBBVWFXo=/ws
Connecting to VM Service at ws://127.0.0.1:63316/aE8KBBVWFXo=/ws
Connected to the VM Service.
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.27.0, on Microsoft Windows [Version 10.0.22631.5624], locale en-IN)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 35.0.1)
[√] Chrome - develop for the web
[!] Visual Studio - develop Windows apps (Visual Studio Build Tools 2019 16.11.46)
    X The current Visual Studio installation is incomplete.
      Please use Visual Studio Installer to complete the installation or reinstall Visual Studio.
[√] Android Studio (version 2024.1)
[√] VS Code (version 1.102.0)
[√] Connected device (3 available)
[√] Network resources

! Doctor found issues in 1 category.
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @itsnits 
I tried the code sample with latest stable and master versions and observed that the `aria-label=Login_email` in intact after setting `ExcludeSemantics` to true and false, both.  

1. With true:

<img width="329" height="31" alt="Image" src="https://github.com/user-attachments/assets/cce6a1b6-0f0f-4ff5-b614-be39473f82c6" />

2. With false:

<img width="299" height="48" alt="Image" src="https://github.com/user-attachments/assets/bf1e2489-ea89-4326-9eb5-2cdc5b7abc66" />

Please try by upgrading to latest sdk as there have been few fixes made related to aria label recently.

**Comment by itsnits** [comment]
Thanks @darshankawar for checking this.

As confirmed by you, I have checked this with the latest stable flutter version 3.32.6. 

1. with excludeSemantics: true, I am getting semantic label correctly as it was in earlier version.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/e5bfa564-c7c2-435b-b75f-4525bb1d29be" />

2. But with excludeSemantics : false, I am getting two input tags. One with aria-label = "Login_email" and the other with aria-label = "Email". I am attaching the screenshot for reference.  So in this case 
 
 - Input is not interactive with "Login_email" label in flutter web automation
 - Input is interactive with "Email" label, but this should work with "Login_email" label.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/029b6424-5c5a-4d88-b7bd-f901e5b641b3" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/ea2539ce-3fcf-48cc-9fc8-f713930cc4a5" />

Please check this.

Flutter doctor output

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.32.6, on Microsoft Windows [Version 10.0.22631.5624], locale en-IN)
[√] Windows Version (11 Enterprise 64-bit, 23H2, 2009)
[√] Android toolchain - develop for Android devices (Android SDK version 35.0.1)
[√] Chrome - develop for the web
[!] Visual Studio - develop Windows apps (Visual Studio Build Tools 2019 16.11.46)
    X The current Visual Studio installation is incomplete.
      Please use Visual Studio Installer to complete the installation or reinstall Visual Studio.
[√] Android Studio (version 2024.1)
[√] VS Code (version 1.102.0)
[√] Connected device (3 available)
[√] Network resources

! Doctor found issues in 1 category

```

**Comment by darshankawar** [comment]
Thanks for the update. Yes, I see the same behavior.

1. With `true`:

<img width="165" height="142" alt="Image" src="https://github.com/user-attachments/assets/b6abbe39-ca33-40be-9275-1319a56a72f8" />

2. With `false`:

<img width="188" height="141" alt="Image" src="https://github.com/user-attachments/assets/1b635be7-753e-4927-a271-f7e5c45819bb" />
<img width="615" height="174" alt="Image" src="https://github.com/user-attachments/assets/8d76c3b7-0df6-464c-a25b-213bf6cb026e" />


```
stable : 3.32.6
master : 3.33.0-1.0.pre-1005
```

**Comment by itsnits** [comment]
Hi @darshankawar,

Thanks for confirming the issue and reproducing it.

I just wanted to check on the next steps. Is there anything else required from my side to assist further?
Also, could you please let me know if a fix is planned for an upcoming release?

Appreciate your support!

**Comment by chunhtai** [comment]
@yjbanov I remember customer:chalk used to do this and run into some issue as well, do you remember how they end up fixing this?

**Comment by itsnits** [comment]
Hi @chunhtai , @mdebbar , Please help me with the update, if any.

**Comment by darshankawar** [comment]
@itsnits Please don't ping team members for updates and read  https://github.com/flutter/flutter/blob/master/docs/contributing/issue_hygiene/README.md#comments
